library/alloc/src/vec/mod.rs

```diff
@@ -152,12 +152,13 @@ mod spec_extend;
 /// assert_eq!(vec, [7, 1, 2, 3]);
 /// ```
 ///
-/// The [`vec!`] macro is provided to make initialization more convenient:
+/// The [`vec!`] macro is provided for convenient initialization:
 ///
 /// ```
-/// let mut vec = vec![1, 2, 3];
-/// vec.push(4);
-/// assert_eq!(vec, [1, 2, 3, 4]);
+/// let mut vec1 = vec![1, 2, 3];
+/// vec1.push(4);
+/// let vec2 = Vec::from([1, 2, 3, 4]);
+/// assert_eq!(vec1, vec2);
```

Comment:
I think we should keep this part. The previous was to show comparison between vec and array. But this one adds an extra step.

Reply:
The goal of this section of the document is to showcase the initialization of Vecs, and I think that demonstrating the `PartialEq<[T; N]>` impl is less relevant than demonstrating `Vec::from` (which was not added by this RFC, though it is relevant to it). The `PartialEq<[T; N]>` impl is already demonstrated in the other examples, and if it is especially important then it should be given its own section.